nilearn/decomposition/multi_pca.py

```diff
@@ -211,7 +221,7 @@ def fit(self, imgs, y=None, confounds=None):
             raise ValueError('Need one or more Niimg-like objects as input, '
                              'an empty list was given.')
         if confounds is None:
-            confounds = itertools.repeat(None, len(imgs))
+            confounds = [None] * len(imgs)  # itertools.repeat(None, len(imgs))
```

Comment:
I don't know if this is the right solution, but I encountered what appeared to be a bug in itertools.repeat. confounds ended up being repeat(None, 0) after being initialized at repeat(None, len(imgs)), and I could not identify why